lms/djangoapps/learner_dashboard/api/v0/views.py

```diff
@@ -0,0 +1,193 @@
+                        1,
+                        2
+                    ],
+                    "is_learner_eligible_for_one_click_purchase": false,
+                    "skus": ["AUD122342"],
+                },
+                "course_data": {
+                    "uuid": "a156a6e2-de91-4ce7-947a-888943e6b12a",
+                    "completed": [],
+                    "in_progress": [],
+                    "not_started": [
+                        {
+                            "key": "edX+DemoX",
+                            "uuid": "fe1a9ad4-a452-45cd-80e5-9babd3d43f96",
+                            "title": "Demonstration Course",
+                            "course_runs": [],
+                            "entitlements": [],
+                            "owners": [],
+                            "image": "",
+                            "short_description": "",
+                            "type": "457f07ec-a78f-45b4-ba09-5fb176520d8a",
+                        }
+                    ],
+                },
+                "certificate_data": [{
+                    "type": "course",
+                    "title": "edX Demo Course",
+                    'url': "/certificates/6e57d3cce8e34cfcb60bd8e8b04r07e0",
+                }],
+                "industry_pathways": [
+                    {
+                        "id": 2,
+                        "uuid": "1b8fadf1-f6aa-4282-94e3-325b922a027f",
+                        "name": "Demo Industry Pathway",
+                        "org_name": "edX",
+                        "email": "edx@edx.com",
+                        "description": "Sample demo industry pathway",
+                        "destination_url": "http://rit.edu/online/pathways/gtx-analytics-essential-tools-methods",
+                        "pathway_type": "industry",
+                        "program_uuids": [
+                            "a156a6e2-de91-4ce7-947a-888943e6b12a"
+                        ]
+                    }
+                ],
+                "credit_pathways": [
+                    {
+                        "id": 1,
+                        "uuid": "86b9701a-61e6-48a2-92eb-70a824521c1f",
+                        "name": "Demo Credit Pathway",
+                        "org_name": "edX",
+                        "email": "edx@edx.com",
+                        "description": "Sample demo credit pathway!",
+                        "destination_url": "http://rit.edu/online/pathways/ritx-design-thinking",
+                        "pathway_type": "credit",
+                        "program_uuids": [
+                            "a156a6e2-de91-4ce7-947a-888943e6b12a"
+                        ]
+                    }
+                ]
+            }
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+
+    permission_classes = (permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS,)
```

Comment:
please see https://edx-internal.slack.com/archives/CD89W7Z7F/p1646343366302809?thread_ts=1645703082.653149&cid=CD89W7Z7F

BTW any idea what is the difference between `SessionAuthenticationAllowInactiveUser` and `SessionAuthentication`?

Reply:
From what I can understand from Robert's comment, I think we can discard the `BearerAuthenticationAllowInactiveUser` class. Even for permissions I believe we will only need `IsAuthenticated` as any authenticated enterprise user should be able to access this endpoint. 

However, I'm still confused b/w  `SessionAuthenticationAllowInactiveUser` and `SessionAuthentication`. `SessionAuthenticationAllowInactiveUser` is an extended class of `SessionAuthentication` which just ensures that users who haven't activated their account (confirmed their email address) are still able to access the endpoint. Only the `user.is_active` check is removed, rest is same. Can this scenario occur for enterprise users?

See this for details: 
https://github.com/openedx/edx-drf-extensions/blob/21d98067fb450dacbe8a0e4b7b3ffe95eba5170c/edx_rest_framework_extensions/auth/session/authentication.py#L6

Reply:
I think we don't need this.

As per my understanding we should start with bare minimum and add/update if need arise.

Reply:
Incorporated. Using `IsAuthenticated` for permissions and `JwtAuthentication` + `JwtAuthentication` for authentication now.